core/src/main/spark-2.1/scala/com/pingcap/tispark/TiConfigConst.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tispark
+
+object TiConfigConst {
+  val PD_ADDRESSES: String = "spark.tispark.pd.addresses"
+  val GRPC_FRAME_SIZE: String = "spark.tispark.grpc.framesize"
+  val GRPC_TIMEOUT: String = "spark.tispark.grpc.timeout_in_sec"
+  val META_RELOAD_PERIOD: String = "spark.tispark.meta.reload_period_in_sec"
+  val GRPC_RETRY_TIMES: String = "spark.tispark.grpc.retry.times"
+  val INDEX_SCAN_BATCH_SIZE: String = "spark.tispark.index.scan_batch_size"
+  val INDEX_SCAN_CONCURRENCY: String = "spark.tispark.index.scan_concurrency"
+  val TABLE_SCAN_CONCURRENCY: String = "spark.tispark.table.scan_concurrency"
+  val ALLOW_AGG_PUSHDOWN: String = "spark.tispark.plan.allow_agg_pushdown"
+  val TASK_PER_SPLIT: String = "spark.tispark.task_per_split"
+  val PD_CLUSTER_ID: String = "spark.tispark.pd.clusterid"
+  val REQUEST_COMMAND_PRIORITY: String = "spark.tispark.request.command.priority"
+  val REQUEST_ISOLATION_LEVEL: String = "spark.tispark.request.isolation.level"
+  val ALLOW_INDEX_READ: String = "spark.tispark.plan.allow_index_read"
+  val COPROCESS_STREAMING: String = "spark.tispark.coprocess.streaming"
+  val UNSUPPORTED_PUSHDOWN_EXPR: String = "spark.tispark.plan.unsupported_pushdown_exprs"
+  val REGION_INDEX_SCAN_DOWNGRADE_THRESHOLD: String = "spark.tispark.plan.downgrade.index_threshold"
+  val KV_TIMEZONE_OFFSET: String = "spark.tispark.request.timezone.offset"
+  val UNSUPPORTED_TYPES: String = "spark.tispark.type.unsupported_mysql_types"
+  val ENABLE_AUTO_LOAD_STATISTICS: String = "spark.tispark.statistics.auto_load"
+  val CACHE_EXPIRE_AFTER_ACCESS: String = "spark.tispark.statistics.expire_after_access"
+  val SHOW_ROWID: String = "spark.tispark.show_rowid"
+  val DB_PREFIX: String = "spark.tispark.db_prefix"
```

Comment:
We should remove this right?

Reply:
@ilovesoup It should still remain because we still cannot distinguish different databases in Hive and TiDB with the same name. The LegacyFirstPolicy/IdentityPolicy was designed to address their priority.

Consider that we both have database `db` in Hive and TiDB, when user calls `use db`, the result will be problematic if there is not a deterministic priority. Using LegacyFirstPolicy will use Hive databases in prior to TiDB databases, thus the database `db` in TiDB will be invisible.

In order to resolve this problem, a DB_PREFIX configuration is added when user wants to minimize its impact by adding a prefix in TiDB databases, "tidb_", for instance, so that they can call `use tidb_db` explicitly. In this circumstance, all databases in TiDB will have the same prefix.